parachain/pallets/ethereum-beacon-client/src/merkleization.rs

```diff
@@ -0,0 +1,613 @@
+    let mut proposer_slashings_vec = Vec::new();
+
+    for proposer_slashing in proposer_slashings.iter() {
+       proposer_slashings_vec.push(get_ssz_proposer_slashing((*proposer_slashing).clone())?);
+    }
+
+    Ok(List::<SSZProposerSlashing, { config::MAX_PROPOSER_SLASHINGS }>::from_iter(proposer_slashings_vec))
+}
+
+pub fn get_ssz_proposer_slashing(proposer_slashing: ProposerSlashing) -> Result<SSZProposerSlashing, MerkleizationError> {
+    let signature1 = Vector::<u8, 96>::from_iter(proposer_slashing.signed_header_1.signature.clone());
+    let signature2 = Vector::<u8, 96>::from_iter(proposer_slashing.signed_header_2.signature.clone());
+
+    Ok(SSZProposerSlashing{
+        signed_header_1: SignedBeaconBlockHeader{
+            message: get_ssz_beacon_header(proposer_slashing.signed_header_1.message.clone())?,
+            signature: signature1,
+        },
+        signed_header_2: SignedBeaconBlockHeader{
+            message: get_ssz_beacon_header(proposer_slashing.signed_header_2.message.clone())?,
+            signature: signature2,
+        },
+    })
+}
+
+pub fn get_ssz_attester_slashings(attester_slashings: Vec<AttesterSlashing>) -> Result<List<SSZAttesterSlashing, { config::MAX_ATTESTER_SLASHINGS } >, MerkleizationError> {
+    let mut attester_slashings_vec = Vec::new();
+
+    for attester_slashing in attester_slashings.iter() {
+        attester_slashings_vec.push(get_ssz_attester_slashing((*attester_slashing).clone())?);
+    }
+
+    Ok(List::<SSZAttesterSlashing, { config::MAX_ATTESTER_SLASHINGS } >::from_iter(attester_slashings_vec))
+}
+
+pub fn get_ssz_attester_slashing(attester_slashing: AttesterSlashing) -> Result<SSZAttesterSlashing, MerkleizationError> {
+    let signature1 = Vector::<u8, 96>::from_iter(attester_slashing.attestation_1.signature.clone());
+    let signature2 = Vector::<u8, 96>::from_iter(attester_slashing.attestation_2.signature.clone());
+
+    let attesting_indices1 = List::<u64, { config::MAX_VALIDATORS_PER_COMMITTEE }>::from_iter(attester_slashing.attestation_1.attesting_indices.clone());
+    let attesting_indices2 = List::<u64, { config::MAX_VALIDATORS_PER_COMMITTEE }>::from_iter(attester_slashing.attestation_2.attesting_indices.clone());
+
+    Ok(SSZAttesterSlashing{
+        attestation_1: SSZIndexedAttestation{
+            attesting_indices: attesting_indices1,
+            data: get_ssz_attestation_data(attester_slashing.attestation_1.data.clone())?,
+            signature: signature1,
+        },
+        attestation_2: SSZIndexedAttestation{
+            attesting_indices: attesting_indices2,
+            data: get_ssz_attestation_data(attester_slashing.attestation_2.data.clone())?,
+            signature: signature2,
+        },
+    })
+}
+
+pub fn get_ssz_checkpoint(checkpoint: Checkpoint) -> Result<SSZCheckpoint, MerkleizationError> {
```

Comment:
Minor Nit: Its a convention to implement [TryInto](https://doc.rust-lang.org/std/convert/trait.TryInto.html) instead of creating ad hoc conversion functions.

But this is fine for now, just keep in mind for the future.

Reply:
Great, thanks for the tip! I'll do a refactor at some point.